### PR TITLE
Add heading no content validation

### PR DESCRIPTION
**Pull request by XiaofuHuang**
[AB#211396](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/211396)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5890)

### CODE REVIEW

File: src/docfx/lib/markdown/validation/ContentValidationExtension.cs [hunk 1]
```diff
@@ -142,5 +148,25 @@ static string GetContainerInlineContent(ContainerInline containerInline)
                 return content.ToString();
             }
         }
+
+        internal class InclusionNode : Node, ICloneable
```
Comment (OsmondJiang) [inline src/docfx/lib/markdown/validation/ContentValidationExtension.cs:152]: private

File: src/docfx/lib/markdown/MarkdigUtility.cs [hunk 2]
```diff
@@ -167,12 +167,18 @@ public static bool IsVisible(this MarkdownObject markdownObject)
                             visiable = visiable || VisibleHtml(line.Slice.Text);
                         }
                         return true;
-                    case HtmlInline htmlInline:
-                        visiable = visiable || VisibleHtml(htmlInline.Tag);
-                        return false;
+                    case ThematicBreakBlock _:
```
Comment (OsmondJiang) [inline src/docfx/lib/markdown/MarkdigUtility.cs:170]: add the new added three cases to MarkdigUtilityTest

Comment (XiaofuHuang) [reply src/docfx/lib/markdown/MarkdigUtility.cs:170]: Heading case has been added. ThematicBreakBlock cannot generated here, it has been tested in validation spec test.

File: src/docfx/lib/markdown/validation/ContentValidationExtension.cs [hunk 3]
```diff
@@ -144,5 +150,25 @@ static string GetContainerInlineContent(ContainerInline containerInline)
...
+
+        private class InclusionNode : ValidationNode, ICloneable
+        {
+            public string? IncludedFilePath { get; set; }
+
+            public InclusionNode()
+            {
+            }
+
+            protected InclusionNode(InclusionNode inclusionNode)
```
Comment (mingwli) [inline src/docfx/lib/markdown/validation/ContentValidationExtension.cs:162]: where's this used?

File: src/docfx/lib/markdown/MarkdigUtility.cs [hunk 4]
```diff
@@ -177,6 +177,7 @@ public static bool IsVisible(this MarkdownObject markdownObject)
                         visible = visible || VisibleHtml(htmlInline.Tag);
                         break;
                     case HeadingBlock headingBlock when headingBlock.Inline is null || !headingBlock.Inline.Any():
+                    case ThematicBreakBlock _:
                         // empty heading
```
Comment (OsmondJiang) [inline src/docfx/lib/markdown/MarkdigUtility.cs:181]: > // empty heading [](start = 23, length = 17)

change or remove the comment

File: src/docfx/lib/markdown/validation/ContentValidationExtension.cs [hunk 5]
```diff
@@ -144,5 +150,25 @@ static string GetContainerInlineContent(ContainerInline containerInline)
                 return content.ToString();
             }
         }
+
+        private class InclusionNode : ValidationNode, ICloneable
```
Comment (OsmondJiang) [inline src/docfx/lib/markdown/validation/ContentValidationExtension.cs:154]: >ICloneable [](start = 54, length = 10)

no need implement the ICloneable interface

File: src/docfx/lib/markdown/validation/ContentValidationExtension.cs [hunk 6]
```diff
@@ -75,22 +83,20 @@ internal static class ContentValidationExtension
...
+                                    nodes.RemoveAt(index);
+                                    nodes.InsertRange(index, documentNodes.Select(node =>
                                     {
-                                        Content = d.Content,
-                                        Level = d.Level,
-                                        SourceInfo = current.SourceInfo,
-                                        HeadingChar = d.HeadingChar,
-                                        RenderedPlainText = d.RenderedPlainText,
-                                        InclusionSourceInfo = d.SourceInfo,
+                                        var newNode = (ValidationNode)node.Clone();
```
Comment (OsmondJiang) [inline src/docfx/lib/markdown/validation/ContentValidationExtension.cs:94]: >(ValidationNode) [](start = 54, length = 16)

no need cast here?

File: src/docfx/lib/markdown/validation/ContentValidationExtension.cs (hunk 3 above)
Comment (XiaofuHuang) [reply src/docfx/lib/markdown/validation/ContentValidationExtension.cs:162]: We need to clone node when inclusion node contains inclusion node. It is used in clone function.

File: src/docfx/lib/markdown/validation/ContentValidationExtension.cs (hunk 6 above)
Comment (XiaofuHuang) [reply src/docfx/lib/markdown/validation/ContentValidationExtension.cs:94]: Because clone return object

**Pull request merged**